cmake-toolchain/README.md

```diff
@@ -0,0 +1,40 @@
+# Cross Compiliation
+ **Notice:** Running the AWS IoT Device Client will incur usage of AWS IoT services, and is likely to incur charges on your AWS account. Please refer the pricing pages for [AWS IoT Core](https://aws.amazon.com/iot-core/pricing/), [AWS IoT Device Management](https://aws.amazon.com/iot-device-management/pricing/), and [AWS IoT Device Defender](https://aws.amazon.com/iot-device-defender/pricing/) for more details.
+
+[*Back To The Main Readme*](../README.md)
+
+## Building from one architecture to the other
+
+**Description**:
+Cross Compiling allows you to compile for a **specific architecture** while on a **different architecture**.  This is extremely useful, as some *(Especially really constrained)* devices won't have the resources to complete the build process.  Even if your devices can complete the build, it is often times much easier to build in a central location (like your developer laptop) and then distribute the artifacts to appropriate devices. 
```

Comment:
Cross Compiling -> Cross-compiling. 
This sentence confused me more about what cross compilation is. I think it needs better wording and introduction here. If someone is new to this concept, it will be hard for him to follow. You can also link some good definition you find from other place.